src/game/logic/object/weapon.cpp

```diff
@@ -1195,15 +1370,236 @@ bool Weapon::Compute_Approach_Target(const Object *source_obj,
     float angle_offset,
     Coord3D &approach_target_pos) const
 {
-    // todo
-    return false;
+    const Coord3D *pos;
+    Coord3D vector;
+
+    if (target_obj != nullptr) {
+        pos = target_obj->Get_Position();
+        g_thePartitionManager->Get_Vector_To(target_obj, source_obj, FROM_BOUNDINGSPHERE_2D, vector);
```

Comment:
This direction is correct? target to source? Not the other way around?

Reply:
Yes this is correct.